### PR TITLE
Outdated Refactor WIP (Do not merge)

### PR DESCRIPTION
**Pull request by timothywangdev**

### CODE REVIEW

File: src/wallets/bitcoin.js [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+      let account = this.walletData.accounts[accountIdx]
+      let { address, xpub } = await this.ledger.getBtcAddresss(accountIdx)
+      account.address = address
+      account.hdWalletVariables.xpub = xpub
+    } else {
+      throw new Error(`Cannot retrieve address for ${walletType}`)
+    }
+  }
+
+  sync = async (progress: any) => {
```
Comment (InVincible2016) [inline src/wallets/bitcoin.js]: progress?: any

File: src/wallets/bitcoin.js [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+    change: number,
+    offset: number,
+    progress: ?Function
+  ): Promise<{ nextIndex: number, addresses: Array<Address> }> => {
+    let gap = 0
+    let addresses: Array<AddressBitcoin> = []
+    let currentIdx = offset
+    let lastUsedIdx = offset - 1
+    while (gap < 5) {
+      console.log(currentIdx)
```
Comment (InVincible2016) [inline src/wallets/bitcoin.js]: remove

File: src/wallets/bitcoin.js [hunk 3]
```diff
@@ -0,0 +1,510 @@
...
+    accountIndex: number,
+    change: number,
+    offset: number,
+    progress: ?Function
+  ): Promise<{ nextIndex: number, addresses: Array<Address> }> => {
+    let gap = 0
+    let addresses: Array<AddressBitcoin> = []
+    let currentIdx = offset
+    let lastUsedIdx = offset - 1
+    while (gap < 5) {
```
Comment (InVincible2016) [inline src/wallets/bitcoin.js]: should we consider changing the gap back to 20?

File: src/containers/ReceiveWalletSelectionContainer.jsx [hunk 4]
```diff
@@ -67,30 +67,30 @@ class ReceiveWalletSelectionContainer extends Component {
...
     }
   }
 
-  onSync = (cryptoType: string) => {
-    const { syncLedgerAccountInfo } = this.props
-    syncLedgerAccountInfo(cryptoType, 0, (index, change) => { this.setState({ syncProgress: { index, change } }) })
+  onSync = () => {
+    let { wallet, walletSelection, transfer } = this.props
+    let accounts = wallet.crypto[transfer.cryptoType]
+    console.log(this.props)
```
Comment (InVincible2016) [inline src/containers/ReceiveWalletSelectionContainer.jsx]: remove

File: src/wallets/ethereum.js [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+          window.ethereum.networkVersion !== networkIdMap[env.REACT_APP_ETHEREUM_NETWORK].toString()
+        ) {
+          throw 'Incorrect Metamask network' // eslint-disable-line
+        }
+        let addresses = await window.ethereum.enable()
+        this.walletData.accounts[accountIdx].address = addresses[0]
+      } else {
+        throw new Error('Metamask not found')
+      }
+    } else {
```
[the remark this answers is not in the document]
Comment (timothywangdev) [reply src/wallets/ethereum.js]: no need to manually retrieve address for drive

File: src/wallets/ethereum.js [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+  clearPrivateKey = (): void => {
+    let accountIdx = 0
+    this.walletData.accounts[accountIdx].privateKey = undefined
+  }
+
+  retrieveAddress = async (): Promise<void> => {
+    let accountIdx = 0
+    let { walletType } = this.walletData
+    if (this.walletData.accounts.length === 0) {
+      this.walletData.accounts.push({
```
[the remark this answers is not in the document]
Comment (timothywangdev) [reply src/wallets/ethereum.js]: retrieveAddress default for first account, need to handle account DNE

File: src/wallets/bitcoin.js (hunk 1 above)
Comment (timothywangdev) [reply src/wallets/bitcoin.js]: fixed

File: src/wallets/bitcoin.js (hunk 2 above)
Comment (timothywangdev) [reply src/wallets/bitcoin.js]: fixed

File: src/wallets/bitcoin.js (hunk 3 above)
Comment (timothywangdev) [reply src/wallets/bitcoin.js]: fixed

File: src/containers/ReceiveWalletSelectionContainer.jsx (hunk 4 above)
Comment (timothywangdev) [reply src/containers/ReceiveWalletSelectionContainer.jsx]: fixed

**Pull request closed without merging**